cirq-core/cirq/ops/moment.py

```diff
@@ -50,6 +50,17 @@ def _default_breakdown(qid: 'cirq.Qid') -> Tuple[Any, Any]:
         return None, qid
 
 
+def _have_equal_gates(op1: 'cirq.Operation', op2: 'cirq.Operation') -> bool:
+    """Compares two operations and returns True if their gates match.
+    Returns:
+        bool: True if both gates are not null, the same, and take in
+           the same number of qubits from their operations.
+    """
+    if op1.gate is None or op2.gate is None:
+        return False
+    return isinstance(op1.gate, type(op2.gate)) and isinstance(op2.gate, type(op1.gate))
```

Comment:
We should compare the gate instances and not have an `isinstance` check. Gate instances with different parameter values will correspond to different gates. 

Eg: `cirq.X ** 0.25` and `cirq.X ** 0.5` are two different gates but your function will return true. 

We can now simply return `op1.gate == op2.gate` as `ParallelGateOperation` is now deprecated, so there's no additional need to compare qubit length (which I had suggested in my previous review).

Reply:
I would like to hear from @dstrain115 as well, seeing as he was concerned with comparing measurement gates and suggested not to use `op1.gate == op2.gate`.

Reply:
Did I recommend isinstance?  I don't remember why and can't find the comment.  My previous comment was regarding comparing the number of qubits, which would fail for gates that can have variable number of qubits (such as measure gates).

Tanuj is correct though, this check will fail for gates with different parameters.

Reply:
Done.